iox_query/src/plan/influxql.rs

```diff
@@ -61,58 +104,583 @@ impl<'a> InfluxQLToLogicalPlan<'a> {
             }
         }
     }
+
+    /// Create a [`LogicalPlan`] from the specified InfluxQL `SELECT` statement.
+    fn select_statement_to_plan(&self, select: SelectStatement) -> Result<LogicalPlan> {
+        // Process FROM clause
+        let plans = self.plan_from_tables(select.from)?;
+
+        // Only support a single measurement to begin with
+        let plan = match plans.len() {
+            0 => Err(DataFusionError::NotImplemented(
+                "unsupported FROM: schema must exist".into(),
+            )),
+            1 => Ok(plans[0].clone()),
+            _ => Err(DataFusionError::NotImplemented(
+                "unsupported FROM: must target a single measurement".into(),
+            )),
+        }?;
+
+        let plan = self.plan_where_clause(select.condition, plan)?;
+
+        // Process and validate the field expressions in the SELECT projection list
+        let select_exprs = self.field_list_to_exprs(&plan, select.fields)?;
+
+        let projected_plan = self.project(plan.clone(), select_exprs.clone())?;
+        let mut combined_schema = (**projected_plan.schema()).clone();
+        combined_schema.merge(plan.schema());
+
+        let plan = project(plan, select_exprs)?;
```

Comment:
I don't understand this code -- it seems to be doing the projection two times (though ignoring the results of the first)

I wonder why isn't it more like

```suggestion
        // project with error checking
        let plan = self.project(plan, select_exprs)?;

```

Reply:
Ahh yes, good catch!

I had originally duplicated this from the `SqlToRel` type:

https://github.com/apache/arrow-datafusion/blob/9bee14ebd39dacbb66a9b1f34cd6494bc6a6be3f/datafusion/sql/src/planner.rs#L1056-L1059

And this code is used to allow the `GROUP BY` (and for SQL also the `HAVING`) clauses to use aliases from the projection. InfluxQL does not support this, so I can remove this redundant code completely.